bundles/org.openhab.binding.windcentrale/src/main/java/org/openhab/binding/windcentrale/internal/handler/WindcentraleAccountHandler.java

```diff
@@ -0,0 +1,156 @@
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.windcentrale.internal.WindcentraleDiscoveryService;
+import org.openhab.binding.windcentrale.internal.api.RequestListener;
+import org.openhab.binding.windcentrale.internal.api.TokenProvider;
+import org.openhab.binding.windcentrale.internal.api.WindcentraleAPI;
+import org.openhab.binding.windcentrale.internal.config.AccountConfiguration;
+import org.openhab.binding.windcentrale.internal.exception.FailedGettingDataException;
+import org.openhab.binding.windcentrale.internal.exception.InvalidAccessTokenException;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link WindcentraleAccountHandler} provides the {@link WindcentraleAPI} instance used by the windmill handlers.
+ *
+ * @author Wouter Born - Initial contribution
+ */
+@NonNullByDefault
+public class WindcentraleAccountHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(WindcentraleAccountHandler.class);
+
+    private final HttpClientFactory httpClientFactory;
+
+    private @Nullable WindcentraleAPI api;
+    private @Nullable Exception apiException;
+    private @Nullable Future<?> initializeFuture;
+
+    private final RequestListener requestListener = new RequestListener() {
+        @Override
+        public void onError(Exception exception) {
+            apiException = exception;
+            logger.debug("API exception occurred");
+            updateThingStatus();
+        }
+
+        @Override
+        public void onSuccess() {
+            if (apiException != null) {
+                apiException = null;
+                logger.debug("API exception cleared");
+                updateThingStatus();
+            }
+        }
+    };
+
+    public WindcentraleAccountHandler(Bridge bridge, HttpClientFactory httpClientFactory) {
+        super(bridge);
+        this.httpClientFactory = httpClientFactory;
+    }
+
+    @Override
+    public void dispose() {
+        Future<?> localFuture = initializeFuture;
+        if (localFuture != null) {
+            localFuture.cancel(true);
+            initializeFuture = null;
+        }
+
+        WindcentraleAPI localAPI = api;
+        if (localAPI != null) {
+            localAPI.dispose();
+            localAPI = null;
+        }
+    }
+
+    public @Nullable WindcentraleAPI getAPI() {
+        return api;
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+
+        initializeFuture = scheduler.submit(() -> {
+            api = initializeAPI();
```

Comment:
If this fails, it seems like the bridge will stay offline - no retries?

Reply:
Typically you'll have a windmill Thing so its handler will continuously poll the API. The bridge goes online/offline based on if those API  callls succeed. That's because the bridge listens to those results using its `RequestListener` implementation.

Reply:
Thanks, I now tracked the flow through these three classes, you are right - nice!